TYPO3.Flow/Classes/TYPO3/Flow/Property/TypeConverter/DateTimeConverter.php

```diff
@@ -143,7 +143,8 @@ public function convertFrom($source, $targetType, array $convertedChildPropertie
         if ($dateAsString === '') {
             return null;
         }
-        if (ctype_digit($dateAsString) && $configuration === null && (!is_array($source) || !isset($source['dateFormat']))) {
+        $configurationForm = $configuration === null ? null : $configuration->getConfigurationValue('TYPO3\Flow\Property\TypeConverter\DateTimeConverter', self::CONFIGURATION_DATE_FORMAT);
```

Comment:
OK, now I understood the code I guess (it was already hard to read before). Can we maybe make it a bit more explicit, e.g. sth like:
```php
$isDateFormatSpecified = $configuration->getConfigurationValue(DateTimeConverter::class, self::CONFIGURATION_DATE_FORMAT) !== null || isset($source['dateFormat']);
if (!$isDateFormatSpecified && ctype_digit($dateAsString)) {
 // ...
```
?